zipkin-lens/src/components/TracePage/SpanDetail/SpanAnnotation.jsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+import moment from 'moment';
+import { withStyles } from '@material-ui/styles';
+import Box from '@material-ui/core/Box';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+
+import { spanAnnotationPropTypes } from '../../../prop-types';
+
+const propTypes = {
+  annotation: spanAnnotationPropTypes.isRequired,
+  classes: PropTypes.shape({}).isRequired,
+};
+
+const style = {
+  cell: {
+    paddingTop: '8px',
+    paddingBottom: '8px',
+  },
+};
+
+const SpanAnnotation = ({ annotation, classes }) => (
+  <Box>
+    <Box fontSize="1.1rem" mb={0.5}>
+      {annotation.value}
+    </Box>
+    <Paper>
+      <Table>
+        <TableBody data-testid="span-annotation--table-body">
+          {
+            [
+              {
+                label: 'Start Time',
+                // moment.js only supports millisecond precision, however our timestamps have
+                // microsecond precision. So we use moment.js to generate the human readable time
+                // with just milliseconds and then append the last 3 digits of the timestamp
+                // which are the microseconds.
+                // NOTE: a.timestamp % 1000 would save a string conversion but drops
+                // leading zeros.
+                value: moment(annotation.timestamp / 1000).format('MM/DD HH:mm:ss.SSS')
+                  + annotation.timestamp.toString().slice(-3),
```

Comment:
I guess the result of this would be something like `08/18 16:13:02.123456`? Since the fractional part of the second (`123456`) is longer than 3, I think we can do a small touch and use `_` to improve readability (so it would be `02.123_456`).

I believe it's common in several programming languages and is coming to JS too: https://github.com/tc39/proposal-numeric-separator#references

Reply:
Also I think it would be useful to extract this functionalitity to something like a global util of this package? (especially for its comment)

Reply:
Thank you for your advice!
Done 🙇